server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -302,6 +303,9 @@ private RestResponse getDetailResponse(NettyRestResponse.Builder responseBuilder
       InfinispanQueryStatisticsInfo.IndexStatistics indexStatistics = getIndexStatistics(cache);
       boolean rehashInProgress = distributionManager != null && distributionManager.isRehashInProgress();
       boolean indexingInProgress = indexStatistics != null && indexStatistics.getReindexing();
+      boolean indexed =  configuration.indexing().enabled();
+      boolean query = indexed ||
+            cache.getCacheConfiguration().encoding().valueDataType().mediaType() == MediaType.APPLICATION_PROTOSTREAM;
```

Comment:
* Should test for both MediaType.APPLICATION_PROTOSTREAM and MediaType.APPLICATION_OBJECT 
* Instead of ```==``` should use ```.match()```
*  cache.getCacheConfiguration().encoding().valueDataType() can be ```null``` if I'm not mistaken

Reply:
It still can cause a NPE

Reply:
ahhh I did not reproduce debugging but I was super tired y